file/nifile.yml

```diff
@@ -0,0 +1,816 @@
+    example:
+      name: Skyline.OneOrMoreErrorsOccurred
+      code: -251040
+      message: One or more errors occurred. See the contained list for details of each error.
+      args: []
+      innerErrors:
+        - name: FileIngestion.IdNotFound
+          code: -251608
+          resourceType: file
+          resourceId: '4afb2ce3741fe11d88838cc9'
+          message: File with id '4afb2ce3741fe11d88838cc9' not found.
+          args: ['4afb2ce3741fe11d88838cc9']
+  FileMetadata:
+    description: File metadata
+    type: object
+    properties:
+      _links:
+        description: >-
+          The links to access and manipulate the file:
+
+          - data: Link to download the file using a GET request
+
+          - delete: Link to delete the file using a DELETE request
+
+          - self: Link to the file's metadata
+
+          - updateMetadata: Link to update the file's metadata using a POST request
+        type: object
+        required: [self]
+        additionalProperties:
+          $ref: '#/definitions/Link'
+        properties:
+          self:
+            $ref: '#/definitions/Link'
+        example:
+          data:
+            href: /nifile/v1/service-groups/Default/files/5b0739cc741fe114f08bd06c/data
+          delete:
+            href: /nifile/v1/service-groups/Default/files/5b0739cc741fe114f08bd06c
+          self:
+            href: /nifile/v1/service-groups/Default/files/5b0739cc741fe114f08bd06c
+          updateMetadata:
+            href: /nifile/v1/service-groups/Default/files/5b0739cc741fe114f08bd06c/update-metadata
+      created:
+        description: The date and time the file was created in the file service
+        type: string
+        format: iso-date-time
+        example: '2018-05-15T18:54:27.519Z'
+      id:
+        description: The file's ID within the service group
+        type: string
+        example: '5afb2ce3741fe11d88838cc9'
+      properties:
+        description: The file's properties
+        type: object
+        additionalProperties:
+          type: string
+        example:
+          Name: myfile.txt
+          MyProperty: Value
+      serviceGroup:
+        description: The service group that owns the file
+        type: string
+        example: Default
+      size:
+        description: The 32-bit file size in bytes. If the value is larger than a 32-bit integer, this value is -1 and the size64 parameter contains the correct value.
+        type: integer
+        example: 7277
+      size64:
+        description: The 64-bit file size in bytes
+        type: integer
+        format: int64
+        example: 2147483648
+  ServiceGroup:
+    type: object
+    properties:
+      name:
+        type: string
+        description: The service group's name
+        example: Default
+      _links:
+        description: >-
+          The links that apply to a service group:
+
+          - deleteFiles: Link to delete multiple files from the service group using a POST
```

Comment:
Here and in all of these: could we delete "a" before the method to make it easier to read?